### PR TITLE
Add step for creating releases for tags

### LINKED ISSUE #59: build git-annex releases with minting a release and uploading builds as its assets
build artifacts disappear from github after some time (30? days or so).  We haven't yet established automated archival of all builds (#2).  @vsoch 's (ab)use of github actions to build singularity containers (https://github.com/singularityhub/singularity-deploy) and uploading them as release assets gave me this now obvious idea that we can orchestrate distribution of builds directly here.  I think it would work nicely for us here: file size limit imposed by github is 2GB and otherwise there seems to be no limits (on number of files or bandwidth) ATM: https://docs.github.com/en/github/administering-a-repository/about-releases

ATM we do not mint github releases here, and have just a collection of the tags.  I guess we need to
- trigger a build for a new release of git-annex (if not yet build/released I guess; and we have already a script to help dispatching those)
- after success build, generate a fresh release and upload all built there artifacts as release asset

### PR DESCRIPTION
**Pull request by jwodder**
Closes #59.

### CODE REVIEW

File: .github/workflows/template/build-{{ostype}}.yaml.j2 [hunk 1]
```diff
@@ -246,6 +246,27 @@ jobs:
...
+          then
+              echo "[INFO] Building a tag; uploading assets to release ..."
+              if gh release list | grep -Fqx "$INPUT_COMMITISH"
+              then
+                  echo "[INFO] Release already exists; uploading assets"
+                  gh release upload "$INPUT_COMMITISH" git-annex[-_]*.*
+              else
+                  echo "[INFO] Creating release"
+                  gh release create "$INPUT_COMMITISH" git-annex[-_]*.*
+              fi
```
Comment (yarikoptic) [inline .github/workflows/template/build-{{ostype}}.yaml.j2:262]: placing possible race condition aside (let's consider unlikely to happen) -- shouldn't `gh release upload` happen also after `gh release create`?  i.e. if eg. linux first finished building, it should first create release and then still upload the build.

**Review by yarikoptic — CHANGES_REQUESTED** [review]
left a single comment I think needing to be addressed

we will be able to submit a request to try it out on some recent release committish right?  I guess we could trigger it for a few recent ones (but not going too far back)

forgot to add a comment: is there an option for `gh release create` to provide release comment/description?  if yes, please add 

This is a non-official release build(s) provided by DataLad team.

or something like that to not make users consider them as the official ones (although I know we have it in the repo description, never hurts to remind).
Adding changelog would have been really cool but I guess too much hassle FWIW.

File: .github/workflows/template/build-{{ostype}}.yaml.j2 (hunk 1 above)
Comment (jwodder) [reply .github/workflows/template/build-{{ostype}}.yaml.j2:262]: [The `create` command](https://cli.github.com/manual/gh_release_create) supports uploading assets as part of the release creation.

**Comment by jwodder** [comment]
@yarikoptic 

> we will be able to submit a request to try it out on some recent release committish right? 

Yes, you can use the [`dispatch-build`](https://github.com/datalad/git-annex/blob/master/.github/workflows/tools/dispatch-build) script for that.  Just specify the name of a tag as an argument.

**Comment by yarikoptic** [comment]
Great, thank you @jwodder.  I will merge. Please trigger a build for the most recent release tag

**Comment by jwodder** [comment]
@yarikoptic I triggered builds for the most recent tag, and [a release](https://github.com/datalad/git-annex/releases/tag/8.20210330) was created by the Ubuntu workflow, but the output from `gh release list` wasn't in the format I was expecting, so the macOS and Windows builds failed to upload their assets.

**Comment by yarikoptic** [comment]
https://github.com/datalad/git-annex/releases/tag/8.20210330 looks neat.  So there is something to fix up and redo? (if release could be deleted/recreated - could do that, if not - than on some older tag)

**Comment by jwodder** [comment]
@yarikoptic I fixed the code, deleted the release, and reran the builds, and now everything works.

**Comment by yarikoptic** [comment]
Great, thank you!

**Pull request merged**